library/src/main/java/io/ona/kujaku/views/KujakuMapView.java

```diff
@@ -511,19 +543,29 @@ private void getMapboxMap() {
                 public void onMapReady(MapboxMap mapboxMap) {
                     KujakuMapView.this.mapboxMap = mapboxMap;
                     mapboxMap.getUiSettings().setCompassEnabled(false);
-
-                    // This disables
-                    addMapScrollListenerAndBoundsChangeEmitterToMap(mapboxMap);
-
-                    // Initial call to the bounds listener
-                    callBoundsChangedListeners();
-                    enableFeatureClickListenerEmitter(mapboxMap);
-
-                    if (droppedPoints != null) {
+                    if (KujakuMapView.this.droppedPoints != null) {
+                        List<io.ona.kujaku.domain.Point> droppedPoints = new ArrayList<>(KujakuMapView.this.droppedPoints);
```

Comment:
Is it necessary to create a new Arraylist ? why not use the existing list  `droppedPoints`.

This looks like duplication

Reply:
This actually avoids duplication of dropped points (dropping a point more than once at the same coordinates).

The global variable `droppedPoints` is used in two different places to determine what points to drop : `onMapReady` and `updateDroppedPoints`.

I think the two happen in different threads so if  `updateDroppedPoints`  is called while `onMapReady`  is dropping points, there is a possibility that they will start dropping the same points since they're operating on the same List.

Creating a new List that `updateDroppedPoints`  cannot alter prevents this.

Makes sense?

Reply:
Still the same issue, if an external call to `io.ona.kujaku.views.KujakuMapView#updateDroppedPoints` acquires the lock first and mapBoxMap is not null, it will drop points and update the global `droppedPoints` then `onMapReady` will acquire the lock and drop the previously dropped points including the ones `io.ona.kujaku.views.KujakuMapView#updateDroppedPoints` has just  dropped.

In the current approach, immediately the map is not null, a new List is created, therefore less risk of duplicate dropped points.